proposals/slottable-request.md

```diff
@@ -0,0 +1,86 @@
+*   Components that fetch data themselves, but provide the ability for the user to customize the rendering of the data.
+
+This proposal describes an interoperable protocol for components to request that their owning context render slotted content into themselves, using data provided via the protocol.
+
+# Goals
+
+*   The ability for users of a component to render requested content using a templating system of their choice, using data supplied by the component.
+*   The ability for requested content to be logically composed by the requesting component into the component's shadow root at a position of its choice.
+*   The ability for requested content to be styled in the same "user" scope the component in question was created in, and using styling mechanisms of the user's choice.
+*   The ability to implement the protocol inside framework-specific wrappers or helpers that expose a user API that looks similar or identical to framework-specific patterns (i.e. "render props" for React)..
+
+
+# Design / Proposal
+
+
+## Protocol Definition
+
+Concretely, the protocol involves dispatching a well-known DOM event to request slotted content (aka "[slottables](https://dom.spec.whatwg.org/#light-tree-slotables)", per the DOM spec) be rendered with the provided `data` argument and assigned to the given `slotName`:
+
+```ts
+export class SlotableRequestEvent<Name extends string, T = unknown> extends Event {
+  readonly data: T | typeof remove;
+  readonly name: Name;
+  readonly slotName: string;
+  constructor(name: Name, data: unknown, key?: string) {
+    super('slottable-request', {bubbles: false, composed: false});
+    this.name = name;
+    this.data = data;
+    this.slotName = key !== undefined ? `${name}.${key}` : name;
+  }
+}
+
+export const remove = Symbol('slottable-request-remove');
+```
+
+When a component wants to render customizable content, it (1) fires a `slottable-request` event for a given slot name, and (2) renders a `&lt;slot>` element into its shadow root corresponding to the requested slot name.
+
+Each time the `slottable-request` event is received, the user should use the provided `data` to render (or update) slotted content into the component that fired the event, where each top-level child rendered for a given request is assigned to the provided `slotName`. The `name` field on the event is used to determine what type of content to render; each unique `name` will generally map to a specific user-managed template. 
+
+Because components may request multiple slotted instances for the same conceptual `name` (for example, in the case of a 'list-item` slot request), the `slotName` may differ from the `name`. Thus, `name` should be used to select which template to render, and `slotName` should be assigned to the given rendered _instance_ and used as a unique key to identify which instance to update for subsequent requests.
```

Comment:
Looks like a malformatted markdown tick.

```suggestion
Because components may request multiple slotted instances for the same conceptual `name` (for example, in the case of a `list-item` slot request), the `slotName` may differ from the `name`. Thus, `name` should be used to select which template to render, and `slotName` should be assigned to the given rendered _instance_ and used as a unique key to identify which instance to update for subsequent requests.
```

Reply:
Thanks!